lfs/proxy_test.go

```diff
@@ -0,0 +1,71 @@
+package lfs
+
+import (
+	"net/http"
+	"net/url"
+	// "os"
+	"testing"
+
+	"github.com/github/git-lfs/vendor/_nuts/github.com/technoweenie/assert"
+)
+
+// --- FAIL: TestProxyFromEnv (0.00s)
+// panic: runtime error: invalid memory address or nil pointer dereference [recovered]
+// panic: runtime error: invalid memory address or nil pointer dereference
+//
+// func TestProxyFromEnv(t *testing.T) {
+// 	os.Setenv("HTTPS_PROXY", "https://proxy-from-env:8080")
+//
+// 	u, err := url.Parse("http://some-host.com:123/foo/bar")
+// 	if err != nil {
+// 		t.Fatal(err)
+// 	}
+// 	req := &http.Request{
+// 		URL:    u,
+// 		Header: http.Header{},
+// 	}
+// 	proxyURL, err := proxyFromGitConfigOrEnvironment(req)
+//
+// 	assert.Equal(t, "proxy-from-env:8080", proxyURL.Host)
+// 	assert.Equal(t, nil, err)
+// }
+
+func TestProxyFromGitConfig(t *testing.T) {
+	oldGitConfig := Config.gitConfig
+	defer func() {
+		Config.gitConfig = oldGitConfig
+	}()
+	Config.gitConfig = map[string]string{"http.proxy": "https://proxy-from-git-config:8080"}
+
+	u, err := url.Parse("http://some-host.com:123/foo/bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+	req := &http.Request{
+		URL:    u,
+		Header: http.Header{},
+	}
+	proxyURL, err := proxyFromGitConfigOrEnvironment(req)
+
+	assert.Equal(t, "proxy-from-git-config:8080", proxyURL.Host)
+	assert.Equal(t, nil, err)
+}
+
+// --- FAIL: TestProxyIsNil (0.00s)
+// panic: runtime error: invalid memory address or nil pointer dereference [recovered]
+// panic: runtime error: invalid memory address or nil pointer dereference
+//
+// func TestProxyIsNil(t *testing.T) {
```

Comment:
This is similar to the other stack trace. However, I think you want to assert that `proxyURL` is nil, not `proxyURL.Host`. In Go, only pointers and interfaces can be nil. Built-in types like integers or strings are initialized to a default value, instead of nil.

```go
var foo string // default value is ""
var age int // default is 0
```

The `proxyFromGitConfigOrEnvironment()` returns a `*url.URL`, which can be nil since it's a pointer to a `url.URL` struct.